src/framework/constraints.ml

```diff
@@ -991,6 +991,39 @@ struct
     if D.is_bot d then raise Deadcode else d
 end
 
+module DeadBranchLifter (S: Spec): Spec =
+struct
+  include S
+
+  let name () = "DeadBranch (" ^ name () ^ ")"
+
+  module Locmap = Deadcode.Locmap
+
+  let dead_branches = function true -> Deadcode.dead_branches_then | false -> Deadcode.dead_branches_else
+
+  let locmap_modify_def d k f h =
+    if Locmap.mem h k then
+      Locmap.replace h k (f (Locmap.find h k))
+    else
+      Locmap.add h k d
+
+  let branch ctx exp tv =
+    if !GU.in_verifying_stage && get_bool "dbg.print_dead_code" then (
+      Locmap.replace Deadcode.dead_branches_cond !Tracing.current_loc exp;
+      try
+        let r = branch ctx exp tv in
+        (* branch is live *)
+        Locmap.replace (dead_branches tv) !Tracing.current_loc false;
```

Comment:
Why do we add live branches to this data structure?

Reply:
I guess this is to only report branches as dead that are dead in all contexts (and also paths)?
If yes, we should comment this here.

Reply:
This functor is now applied outside of `PathSensitive2` (previously it was inside in base), but path-wise it makes no difference because `PathSensitive2` aggregates the `Deadcode` exceptions correctly for us.
So it's just for multiple contexts in deed.